metabulo/plot.py

```diff
@@ -17,3 +18,17 @@ def make_box_plot(frame):
     output = io.BytesIO()
     FigureCanvas(fig1).print_png(output)
     return output.getvalue()
+
+
+def pca(metabolites, metadata, x_component_index, y_component_index):
+    n_components = max(x_component_index, y_component_index) + 1
```

Comment:
is this the same as `n_components == min(n_samples, n_features)`  ?

Reply:
No.  This is just more efficient because it only computes the minimal number of singular vectors (SVD is rather slow especially if we get into hundreds of metabolites).

With the code you wrote, it was always returning the largest principal component vs. the second largest.  In that case, you only needed the first two components (the two largest singular vectors).  I modified the endpoint to scatter plot the samples projected onto arbitrary components in each axis.

BTW if we ever do anything like the PCA overview plot from metaboanalyst, it will be cheaper just to return all combinations in one call returning a (ncomponents x ncomponents x nsamples) array.

Reply:
:+1: